_examples/basic/2-realtime-feed/producer/main.go

```diff
@@ -33,42 +34,14 @@ func main() {
 	}
 	defer publisher.Close()
 
-	messagePublished := make(chan struct{})
-	allMessagesPublished := make(chan struct{})
-
-	go func() {
-		publishedMessages := 0
-		for range messagePublished {
-			publishedMessages++
-
-			if publishedMessages%1000 == 0 {
-				logger.Info("messages left", watermill.LogFields{"count": messagesToPublish - publishedMessages})
-			}
-			if publishedMessages >= messagesToPublish {
-				allMessagesPublished <- struct{}{}
-				break
-			}
-		}
-	}()
-
-	workerMessages := make(chan struct{})
+	messageStream := make(chan *message.Message, workers)
+	workersGroup := &sync.WaitGroup{}
```

Comment:
I think you can just do `wg.Add(workers)`, as using `Add` and `Wait` concurrently is considered a race.

Reply:
637a8b3c7b5510560fee8b42ecd710f5f06a8d54